packages/worker/src/runtime.ts

```diff
@@ -377,11 +378,20 @@ export class Runtime {
   /**
    * Used by Workers to register for shutdown signals
    *
+   * Returns false if a shutdown has already been requested. In that case, the Worker should
+   * immediately initiate its shutdown.
+   *
    * Hidden in the docs because it is only meant to be used internally by the Worker.
+   * @returns true if callback was registered, or false if shutdown has already been requested
    * @hidden
    */
-  public registerShutdownSignalCallback(callback: () => void): void {
-    this.shutdownSignalCallbacks.add(callback);
+  public registerShutdownSignalCallback(callback: () => void): boolean {
```

Comment:
What's the benefit of eagerly returning `false` here?

Reply:
It seems like it would save some complexity on the worker side to always use a callback.